src/main/java/frc/robot/maps/subsystems/SwerveDriveMap.java

```diff
@@ -0,0 +1,289 @@
+        io.rearLeftSteeringCurrentAmps = this.rearLeft.getSteeringMotor().getCurrentAmps();
+        io.rearLeftSteeringTempC = this.rearLeft.getSteeringMotor().getTemperatureC();
+        this.rearLeft.setDesiredState(io.rearLeftDesired);
+
+        io.rearRightDrivePositionMeters = this.rearRight.getDistance();
+        io.rearRightVelocityMetersPerSec = this.rearRight.getState().speedMetersPerSecond;
+        io.rearRightPodAngle = this.rearRight.getAngle().getDegrees();
+        io.rearRightDriveCurrentAmps = this.rearRight.getDriveMotor().getCurrentAmps();
+        io.rearRightDriveTempC = this.rearRight.getDriveMotor().getTemperatureC();
+        io.rearRightSteeringCurrentAmps = this.rearRight.getSteeringMotor().getCurrentAmps();
+        io.rearRightSteeringTempC = this.rearRight.getSteeringMotor().getTemperatureC();
+        this.rearRight.setDesiredState(io.rearRightDesired);
+
+        io.gyroYawPositionDegrees = this.gyro.getAngle();
+    }
+
+    public static class Data implements LoggableInputs {
+        public double frontLeftDrivePositionMeters;
+        public double frontLeftVelocityMetersPerSec;
+        public double frontLeftPodAngle;
+        public double[] frontLeftDriveCurrentAmps;
+        public double[] frontLeftDriveTempC;
+        public double[] frontLeftSteeringCurrentAmps;
+        public double[] frontLeftSteeringTempC;
+        public SwerveModuleState frontLeftDesired = new SwerveModuleState();
+
+        public double frontRightDrivePositionMeters;
+        public double frontRightVelocityMetersPerSec;
+        public double frontRightPodAngle;
+        public double[] frontRightDriveCurrentAmps;
+        public double[] frontRightDriveTempC;
+        public double[] frontRightSteeringCurrentAmps;
+        public double[] frontRightSteeringTempC;
+        public SwerveModuleState frontRightDesired = new SwerveModuleState();
+
+        public double rearLeftDrivePositionMeters;
+        public double rearLeftVelocityMetersPerSec;
+        public double rearLeftPodAngle;
+        public double[] rearLeftDriveCurrentAmps;
+        public double[] rearLeftDriveTempC;
+        public double[] rearLeftSteeringCurrentAmps;
+        public double[] rearLeftSteeringTempC;
+        public SwerveModuleState rearLeftDesired = new SwerveModuleState();
+
+        public double rearRightDrivePositionMeters;
+        public double rearRightVelocityMetersPerSec;
+        public double rearRightPodAngle;
+        public double[] rearRightDriveCurrentAmps;
+        public double[] rearRightDriveTempC;
+        public double[] rearRightSteeringCurrentAmps;
+        public double[] rearRightSteeringTempC;
+        public SwerveModuleState rearRightDesired = new SwerveModuleState();
+
+        public double gyroYawPositionDegrees = 0.0;
+
+        @Override
+        public void toLog(LogTable table) {
+            // frontLeft Desired state (velocity/angle)
+            table.put("FrontLeftDesiredVelocityMetresPerSec",
```

Comment:
You know what would be cool? [Subtables](https://github.com/Mechanical-Advantage/AdvantageKit/blob/main/junction/core/src/org/littletonrobotics/junction/LogTable.java#L80)